broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java

```diff
@@ -3520,23 +3521,46 @@ public MessageEnqueueRecord getEnqueueRecord()
                                                                           destination,
                                                                           parseSelector(selector),
                                                                           limit);
-        _virtualHost.executeTransaction(transaction);
+        executeTransactionAndLogResults("copyMessages", destination.getName(), transaction);
         return transaction.getModifiedMessageIds();
 
     }
 
+    private void executeTransactionAndLogResults(final String msg,
```

Comment:
The entire method implementation can be simplified to something like

`

try
{
    _virtualHost.executeTransaction(transaction);
}
finally
{
    logOperation(String.format("%s  : %s: %s: %d",
                                      operationName,
                                       getName(),
                                       ofNullable(destinationName).orElse(""),
                                       transaction.getModifiedMessageIds().size()));
}
`

Reply:
As per another comment this method is removed